From 6e1500075fc0c2a33e94788b688c28ec0c90eb81 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 11 Sep 2024 06:52:25 +0300
Subject: [PATCH] [FSW-419] fs1: rma: add efuse check for rma enabled

This patch adds check if rma is enabled in efuse and
stores the info in persistent section.

Change-Id: I9fdb12c37c47a58bc0cd8dc09c963e421cb3f555
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 8cd5574..f5dbd8b 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -28,6 +28,7 @@
 	.tpm_extend_done = 0,
 	.tpm_next_pcr_index = 0,
 	.tpm_spi = NULL,
+	.sec_data = {0},
 };
 
 struct boot_rom_manifest_s __attribute__((section (".rom_manifest_section")))boot_rom_manifest;
diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index a375802..4a4be86 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -380,3 +380,13 @@
 
 	return 0;
 }
+
+void boot_rom_check_rma_enabled(void)
+{
+	boot_rom_global.sec_data.rma_enabled = 0;
+
+	/* TODO: Check if TPM owner is removed */
+
+	if (boot_rom_global.config2_fuse & EFUSE_SYSCFG2_RMA_MODE)
+		boot_rom_global.sec_data.rma_enabled = 1;
+}
diff --git a/common/common.c b/common/common.c
index 00c8771..b232389 100644
--- a/common/common.c
+++ b/common/common.c
@@ -116,11 +116,6 @@
 	/* Empty Body */
 }
 
-void boot_rom_check_rma_enabled(void)
-{
-	/* Empty Body */
-}
-
 void boot_rom_check_recovery(void)
 {
 	bool override = true;
